grr/server/grr_response_server/gui/ui/lib/api/api_interfaces.ts

```diff
@@ -10,22 +10,115 @@
 
 export declare interface AnyObject {
   '@type'?: string;
-  [key: string]: undefined|null|string|number|boolean|object;
+  [key: string]: undefined | null | string | number | boolean | object;
+}
+
+/** Api RdfValueFieldDescriptor proto mapping. */
+export declare interface RdfValueFieldDescriptor {
+  readonly name?: string;
+  readonly doc?: string;
+  readonly friendlyName?: string;
+}
+
+/** Api RdfValueDescriptor proto mapping. */
+export declare interface RdfValueDescriptor {
+  readonly name?: string;
+  readonly doc?: string;
+  readonly fields?: ReadonlyArray<RdfValueFieldDescriptor>;
+}
+
+/** ApiAllRdfValues proto mapping. */
+export declare interface ApiAllRdfValues {
+  readonly items?: ReadonlyArray<RdfValueDescriptor>;
 }
 
 /**
  * int64, fixed64, and DecimalString in Protobuf are serialized as string in
  * JSON because JS loses precision for big numeric types. During
  * deserialization, both decimal strings and numbers are accepted.
  */
-export type DecimalString = string|number;
+export type DecimalString = string | number;
+
+/**
+ * ApiUser protomapping.
+ */
+export declare interface ApiUser {
+  readonly username?: string;
+  readonly lastLogon?: string;
+  readonly fullName?: string;
+  readonly homedir?: string;
+  readonly uid?: number;
+  readonly gid?: number;
+  readonly shell?: string;
+}
+
+/**
+ * Network address proto mapping
+ */
+export declare interface ApiNetworkAddress {
+  readonly addressType?: string;
+  readonly packedBytes?: string;
+}
+
+/**
+ * Network Interface proto mapping
+ */
+export declare interface ApiInterface {
+  readonly macAddress?: string;
+  readonly ifname?: string;
+  readonly addresses?: ReadonlyArray<ApiNetworkAddress>;
+}
 
 /**
  * KnowledgeBase proto mapping.
  */
 export declare interface ApiKnowledgeBase {
   readonly fqdn?: string;
+  readonly timeZone?: string;
+
   readonly os?: string;
+  readonly osMajorVersion?: number;
+  readonly osMinorVersion?: number;
+}
+
+/**
+ * ApiClientInformation proto mapping.
+ */
+export declare interface ApiClientInformation {
+  readonly clientName?: string;
+  readonly clientVersion?: number;
+  readonly revision?: number;
+  readonly buildTime?: string;
+  readonly clientBinaryName?: string;
+  readonly clientDescription?: string;
+  readonly labels?: string[];
+}
+
+/** ApiWindowsVolume proto mapping. */
+export declare interface ApiWindowsVolume {
+  readonly attributesList?: string[];
+  readonly driveLetter?: string;
+  readonly driveType?: string;
+}
+
+/** ApiUnixVolume proto mapping. */
+export declare interface ApiUnixVolume {
+  readonly mountPoint?: string;
+  readonly options?: string;
+}
+
+/** ApiVolume proto mapping. */
+export declare interface ApiVolume {
+  readonly name?: string;
+  readonly devicePath?: string;
+  readonly fileSystemType?: string;
+  readonly totalAllocationUnits?: number;
```

Comment:
Surprise: Protobuf fields that are `uint64` are actually rendered as string in JSON. You can check in proto/grr_response_proto/sysinfo.proto and the other Protobuf definition files which fields have what datatype. In the type definition, use `DecimalString` as type for these fields. In the API->Model translation layer, please parse the numbers (preferrably as `BigInt` to maximize precision for numbers larger than 9P.

Reply:
Why are they rendered as string? Is this Protobuf specific or does this happen in our implementation only?

Reply:
Both integers and floating point numbers are represented by the [Number](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number) type in JavaScript. When integers get too big, JavaScript stops representing them with perfect precision and silently uses floating point numbers instead. Only integers up to Number.MAX_SAFE_INTEGER (and Number.MIN_SAFE_INTEGER) are guaranteed to be stored without loss in accuracy.

MAX_SAFE_INTEGER is `2^53 - 1` (~9P) which is less than the max uint64 (`2^64 - 1`). That's why Protobuf serializes certain numeric types as JSON string instead of number. See: https://developers.google.com/protocol-buffers/docs/proto3#json